ci/update-checker/src/main/scala/com/twitter/intellij/updatechecker/PlatformUpdateChecker.scala

```diff
@@ -0,0 +1,26 @@
+package com.twitter.intellij.updatechecker
+
+import org.jsoup.Jsoup
+import org.virtuslab.ideprobe.dependencies.IntelliJVersion
+
+import scala.jdk.CollectionConverters._
+
+object PlatformUpdateChecker {
+  
+  private val snapshotsPage = "https://www.jetbrains.com/intellij-repository/snapshots"
+  
+  private val latestIdeaSnapshotSelector = "h2:containsOwn(com.jetbrains.intellij.idea) + table > tbody > tr:eq(1) > td:lt(3)"
+
+  def latestUpdateVersion(): IntelliJVersion = {
+    println(s"Looking for snapshots at $snapshotsPage")
+    val doc = Jsoup.connect(snapshotsPage).get
+    // See https://jsoup.org/cookbook/extracting-data/selector-syntax for explanation
+    val latestSnapshotRows = doc.select(latestIdeaSnapshotSelector).asScala
+    println("Latest snapshot properties:")
+    latestSnapshotRows.foreach(row => println(s" $row"))
+    
+    val buildNumber = latestSnapshotRows(2).text
```

Comment:
Can we reduce magic numbers in this file?

Reply:
These magic numbers come from the structure of the HTML page. There are more of them embedded in the jsoup selector string.

The bit of information that we have to extract is located in the third cell of the first row of the table that is preceded by a heading that says "com.jetbrains.intellij.idea". These HTML tags have no useful attributes such as class. I don't see how the correct element can be extracted without magic numbers and assumptions about the page structure.